contrib/platform/src/com/sun/jna/platform/win32/COM/COMLateBindingObject.java

```diff
@@ -559,21 +559,19 @@ protected void setProperty(String propertyName, short value) {
 
     /**
      * Sets the property.
-     * <p>
-     * <i>Implementation note:</i> the string is wrapped as a BSTR value, that is
-     * allocated using {@link com.sun.jna.platform.win32.OleAuto#SysAllocString} and
-     * will no longer be accessible to the user to free using
-     * {@link com.sun.jna.platform.win32.OleAuto#SysFreeString}. Consider using
-     * {@link #setProperty(String, VARIANT)} to allow later clearing of the VARIANT.
-     * </p>
      *
      * @param propertyName
      *            the property name
      * @param value
      *            the value
      */
     protected void setProperty(String propertyName, String value) {
-        this.oleMethod(OleAuto.DISPATCH_PROPERTYPUT, null, propertyName, new VARIANT(value));
+        VARIANT wrappedValue = new VARIANT(value);
+        try {
+            this.oleMethod(OleAuto.DISPATCH_PROPERTYPUT, null, propertyName, wrappedValue);
```

Comment:
This change looks good.  However, something to consider.  We already have (next method below) a `setProperty(String propertyName, VARIANT value)` method that we could simply delegate to and accomplish the same thing.  Which prompted my next thought...

Looking at the docs on [`VariantClear()`](https://docs.microsoft.com/en-us/windows/win32/api/oleauto/nf-oleauto-variantclear) I see that it applies to more than just `BSTR`:
> If the vtfield is `VT_BSTR`, the string is freed. If the vtfield is `VT_DISPATCH`, the object is released. If the vt field has the `VT_ARRAY` bit set, the array is freed.

Scanning this file I see a `setProperty(String propertyName, Dispatch value)`.  Does this need similar treatment?  

And rather than duplicating the try/finally logic from here to that method, why not put the try/finally object in the `VARIANT` version of `setProperty()` and add an overload for an optional `boolean` parameter (e.g., `clear`) similar to `OaIdlUtil.toPrimitiveArray()`'s `destruct` parameter or `Convert.toJavaObject()`'s `freeValue` parameter.

Reply:
I think the biggest issue with COM in JNA is the overcomplicated binding for the COM invokers and the VARIANT binding. If the basic interface would be closer to the metal, it would be easier to reason with it.

I would not tinker with a VARIANT I did not create - the user created it, the user can clear it (or not if it is appropriate). The String -> BSTR case is different. In that case we explicitly create a reference counted object (the BSTR) and the user has no control over this.

I saw the Dispatch and IDispatch cases, but also in that case, we just wrap the VARIANT around an existing object without tinkering with the reference count, so the VARIANT is only the structure, nothing more.